### PR TITLE
Add support for Razer Naga Trinity

### PR DESCRIPTION
**Pull request by vincenttaglia**
The static color changing is pretty bloated right now because one of the many packets sent before the static color changing command is some sort of mode switcher that needs to be sent after initialization before color changing command payloads are sent or they won't do anything. I'm planning on narrowing it down to the one required command and executing that only on mouse initialization.

### CODE REVIEW

**Comment by vincenttaglia** [comment]
Every feature available with the Naga and programmed here is working as intended. I think there is enough functionality and polish to merge with the main at this point.

**Comment by vincenttaglia** [comment]
I trimmed the fat in the set static effect method, and refactored the code to look more in line with the rest of the program. I think this branch is at a good place to merge with master.

**Comment by RobertDziarnowski** [comment]
What is required to get it merged?

**Comment by vincenttaglia** [comment]
@robdz not sure about what it takes to get merged, but I have instructions on how to compile the working branch yourself [here](https://github.com/openrazer/openrazer/issues/541#issuecomment-531525329).

File: README.md [hunk 1]
```diff
@@ -82,6 +82,7 @@ The below devices are fully feature supported by OpenRazer which means all avail
 | Razer Abyssus 2000                            |  1532:005E  |
 | Razer Lancehead Tournament Edition            |  1532:0060  |
 | Razer DeathAdder Essential                    |  1532:006E  |
+| Razer Naga Trinity                            |  1532:0067  |
```
Comment (z3ntu) [inline README.md:84]: Please move this line one up, to have them sorted

File: driver/razerchromacommon.c [hunk 2]
```diff
@@ -1193,6 +1193,36 @@ struct razer_report razer_chroma_misc_set_orochi2011_poll_dpi(unsigned short pol
     return report;
 }
 
+/**
+ * Set the Naga Trinity to "Static" effect
+ *
+ * Status Trans Packet Proto DataSize Class CMD Args
+ * 00     1f    0000   00    0e       0f    03  000000000200ffaa00ffaa00ffaa | SET LED MATRIX Effect (NOSTR, Main light (0x00), Static 0x00 (Unknown), Static 0x00 (Unknown), RGB 0x00FFAA, RGB 0x00FFAA, RGB 0x00FFAA)
+ * 00     1f    0000   00    0e       0f    03  0000000002aa00ffaa00ffaa00ff | SET LED MATRIX Effect (NOSTR, Main light (0x00), Static 0x00 (Unknown), Static 0x00 (Unknown), RGB 0xAA00FF, RGB 0xAA00FF, RGB 0xAA00FF)
```
Comment (z3ntu) [inline driver/razerchromacommon.c]: I'm assuming this comment isn't correct so remove it please

File: driver/razermouse_driver.c [hunk 3]
```diff
@@ -443,6 +443,23 @@ static ssize_t razer_attr_write_mode_static(struct device *dev, struct device_at
             report.transaction_id.id = 0xff;
             break;
 
+        case USB_DEVICE_ID_RAZER_NAGA_TRINITY:
+            ;
```
Comment (z3ntu) [inline driver/razermouse_driver.c]: Stray semicolon

File: driver/razermouse_driver.c [hunk 4]
```diff
@@ -443,6 +443,23 @@ static ssize_t razer_attr_write_mode_static(struct device *dev, struct device_at
...
+            report = get_razer_report(0x0f, 0x02, 0x06);
+            report.arguments[0] = 0x00;
+            report.arguments[1] = 0x00;
+            report.arguments[2] = 0x08;
+            report.arguments[3] = 0x00;
+            report.arguments[4] = 0x00;
+            report.arguments[5] = 0x00;
+            report.transaction_id.id = 0x1f;
+
+            razer_send_payload(usb_dev, &report);
```
Comment (z3ntu) [inline driver/razermouse_driver.c:457]: I'm not happy about this section, but I'm not sure where to move it if it's really required :(

**Review by z3ntu — CHANGES_REQUESTED** [review]
See comments

File: driver/razermouse_driver.c (hunk 3 above)
Comment (vincenttaglia) [reply driver/razermouse_driver.c]: At some point the code wouldn't compile without that semicolon but it doesn't seem to be an issue now.

File: driver/razermouse_driver.c (hunk 4 above)
Comment (vincenttaglia) [reply driver/razermouse_driver.c:457]: Yea, it's completely different than other payloads that have established methods. I wasn't sure where to put it either.

File: driver/razerchromacommon.c (hunk 2 above)
Comment (vincenttaglia) [reply driver/razerchromacommon.c]: Done

File: README.md (hunk 1 above)
Comment (vincenttaglia) [reply README.md:84]: Done

**Comment by mateuszkoros** [comment]
Hey, what is the status? I see this PR has been stuck for some time.

**Comment by poirierdag** [comment]
> Hey, what is the status? I see this PR has been stuck for some time.

I 2nd that question, can't wait to be able to use my mouse.

**Comment by vincenttaglia** [comment]
Waiting on a re-review from z3ntu

**Review by z3ntu — APPROVED** [review]
LGTM

I've squashed the commits and added a Co-authored-by tag for you, thank you very much for pushing this forward!

**Pull request merged**